src/components/calcite-input/calcite-input.tsx

```diff
@@ -298,6 +315,7 @@ export class CalciteInput {
         tabIndex={this.disabled ? -1 : null}
         type={this.type}
         value={this.value}
+        valueAsNumber={this.valueAsNumber}
```

Comment:
I'm wondering if we want to set `valueAsNumber` here in addition to `value`.  What happens when the two values are not equivalent?  Which one wins?

I'm wondering why not just implement a `valueAsNumber` property getter in `calcite-input` that returns `this.childEl.valueAsNumber` directly, since that property is reflected by the input's `value` property rather than creating a separate `valueAsNumber` property on calcite-input?

Reply:
Looks like part of the ask here is to be able to set the `valueAsNumber` property too.  In that case you could just implement a setter function in the calcite-input class that takes the incoming value and sets it directly on `this.childEl`.

Reply:
I just realized that `valueAsNumber` is only a property, not an attribute of `HTMLInputElement` so I would advocate against creating a separate Stencil `@Prop` specifically to store this value on `calcite-input`.  What is the use case for someone setting `valueAsNumber` directly rather than just `value`?

Reply:
What if we just expose childEl as a readOnly property? Would that allow users to set valueAsNumber easier?

Reply:
I'm wondering why a user needs to set `valueAsNumber` instead of just `value`?  We could expose `childEl` read only, that would allow any property to be accessed.  I'm wondering what the use case is for this still, asking that question on the original issue to get further background on this.

Reply:
I could just remove setting the valueAsNumber here.

Reply:
Yeah I think that makes sense.  Based on @codylawson's comment sounds like the property just needs to be read from not written to.

Reply:
Honestly, I think i'll close this and respond in the issue that we should probably expose childEl so that a user can use and set valueAsNumber as well as valueAsDate. What do you think?